tests/test_request_header_authentication.py

```diff
@@ -0,0 +1,31 @@
+from unittest import mock  # noqa
+
+import pytest
+from werkzeug.test import EnvironBuilder
+
+from notifications_utils.request_helper import CustomRequest, _check_proxy_header_secret
+
+
+@pytest.mark.parametrize('test_input,expected', [
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['right_key', 'old_key']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['right_key']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['right_key', '']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'My-New-Header', 'header_value': 'right_key', 'secrets': ['right_key', '']}, (True, 'Key used: 1')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['', 'right_key']}, (True, 'Key used: 2')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['', 'old_key', 'right_key']}, (True, 'Key used: 3')),  # noqa
+    ({'secrets': ['old_key', 'right_key']}, (False, 'Header missing')),
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': '', 'secrets': ['right_key', 'old_key']}, (False, 'Header exists but is empty')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'right_key', 'secrets': ['', None]}, (False, 'Secrets are not configured')),  # noqa
+    ({'header_name': 'X-Custom-Forwarder', 'header_value': 'wrong_key', 'secrets': ['right_key', 'old_key']}, (False, "Header didn't match any keys")),  # noqa
```

Comment:
i'd make the parametrize have three values: `headers, secrets, expected_output`. then the first arg would be `{'X-Custom-Forwarder': 'wrong_key'}`, second would be `['right_key', 'old_key']`, and third would be as it is.

You can then get rid of the logic in the test - and just call `builder.headers.update(headers)`

Reply:
Done